### PR TITLE
[ANOMALY DETECTION] Add Anomaly Detection task

### PR DESCRIPTION
**Pull request by djdameln**
## Description
This PR adds the Anomaly Detection task implementation.

[BMM](https://ci.iotg.sclab.intel.com/job/IMPT/job/IMPTOps/27302/)  ✅ 

## Screenshots from SC
Screenshot 2022-03-23 at 10 19 54
Screenshot 2022-03-23 at 10 20 15

### CODE REVIEW

**Comment by djdameln** [comment]
run ote_sdk tests

**Comment by djdameln** [comment]
run ote_sdk tests

**Comment by djdameln** [comment]
run ote_sdk tests

**Comment by samet-akcay** [comment]
run ote_sdk tests

**Comment by samet-akcay** [comment]
run ote_sdk tests

File: tests/ote_cli/common.py [hunk 1]
```diff
@@ -408,8 +417,11 @@ def nncf_eval_testing(template, root, ote_dir, args, threshold):
         evaluated_performance = json.load(read_file)
 
     for k in trained_performance.keys():
-        assert abs(trained_performance[k] - evaluated_performance[k]) / trained_performance[k] <= threshold, \
-            f"{trained_performance[k]=}, {evaluated_performance[k]=}"
+        assert (
+            abs(trained_performance[k] - evaluated_performance[k])
+            / trained_performance[k]
```
Comment (Ilya-Krylov) [inline tests/ote_cli/common.py]: why don't you add `+ 1e-10` here ?

Comment (Ilya-Krylov) [reply tests/ote_cli/common.py]: BTW thanks for doing this

Comment (samet-akcay) [reply tests/ote_cli/common.py]: no reason, probably just missed it :D

**Comment by Ilya-Krylov** [comment]
resolve conflicts please

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Pull request merged**